Content.Shared/Silicons/StationAi/SharedStationAiSystem.Airlock.cs

```diff
@@ -1,25 +1,117 @@
 using Content.Shared.Doors.Components;
 using Robust.Shared.Serialization;
+using Content.Shared.Electrocution;
+using Content.Shared.Popups;
+using Content.Shared.Power.Components;
+using Robust.Shared.Audio;
 
 namespace Content.Shared.Silicons.StationAi;
 
 public abstract partial class SharedStationAiSystem
 {
+    private static readonly SoundPathSpecifier AirlockOverchargeDisabled = new("/Audio/Machines/airlock_overcharge_on.ogg");
+    private static readonly SoundPathSpecifier AirlockOverchargeEnabled = new("/Audio/Machines/airlock_overcharge_off.ogg");
+
     // Handles airlock radial
 
     private void InitializeAirlock()
     {
         SubscribeLocalEvent<DoorBoltComponent, StationAiBoltEvent>(OnAirlockBolt);
+        SubscribeLocalEvent<AirlockComponent, StationAiEmergencyAccessEvent>(OnAirlockEmergencyAccess);
+        SubscribeLocalEvent<ElectrifiedComponent, StationAiElectrifiedEvent>(OnElectrified);
     }
 
+    /// <summary>
+    /// Attempts to bolt door. If wire was cut (AI or for bolts) or its not powered - notifies AI and does nothing.
+    /// </summary>
     private void OnAirlockBolt(EntityUid ent, DoorBoltComponent component, StationAiBoltEvent args)
     {
+        if (!TryComp<StationAiWhitelistComponent>(ent, out var whiteList))
+        {
+            return;
+        }
+
+        if (!whiteList.Enabled || component.BoltWireCut || !component.Powered)
+        {
+            _popup.PopupClient(Loc.GetString("ai-device-not-responding"), args.User, PopupType.MediumCaution);
+            return;
+        }
+
```

Comment:
The latter 3 checks should be handled via a setbolts overload not here because other callers will forget to do this.

Enabled as mentioned should be handled in the BUI message attempt.

Reply:
Need to add TrySetBolt to send popup in case it fails

Reply:
fixed